src/components/engine/create-engine-instance.tsx

```diff
@@ -144,75 +152,169 @@ export const CreateEngineInstanceButton = ({
 };
 
 const CreateCloudHostedEngineModal = ({
-  hasValidPayment,
   disclosure,
-  onConfirm,
-  onAddPaymentMethod,
+  onClickAddToPlan,
 }: {
-  hasValidPayment: boolean;
   disclosure: UseDisclosureReturn;
-  onConfirm: () => void;
-  onAddPaymentMethod: () => void;
+  onClickAddToPlan: (tier: EngineTier) => void;
 }) => {
-  const addToPlan = async () => {
-    if (hasValidPayment) {
-      await onConfirm();
-    } else {
-      await onAddPaymentMethod();
+  const trackEvent = useTrack();
+  const router = useRouter();
+
+  const onClick = async (tier: EngineTier) => {
+    if (tier === "ENTERPRISE") {
+      trackEvent({
+        category: "engine",
+        action: "update-relayer",
+        label: "error",
```

Comment:
this seems like the wrong data to track here?

Reply:
@farhanW3 can you update this, too - pretty sure it shouldn't be `update-relayer` action